adcpipeline/pipeline.py

```diff
@@ -133,3 +139,37 @@ def run(self) -> None:
         logger.info(f'Running pipeline using the following settings: {self.method_settings}')
         for method in self.method_list:
             method()
+        if self.path is not None:
+            self.df.to_pickle(self.path)
+
+    def run_or_load(self, from_step: int = None) -> None:
+        """
```

Comment:
Check if df is also not None?

Reply:
I don't see why that matters at this stage. It is checked if the cache file exists.

Reply:
`self.df.to_pickle(step_path)` won't work if it's None, so I thought better check at the beginning of the method (caching doesn't make sense if df is None)